app/channels/room_channel.rb

```diff
@@ -13,16 +15,32 @@ def send_message(data)
     chat = get_chat(room_id)
     raise 'No message!' if message.blank?
 
-    Message.create!(chat: chat, user: sender, text: message)
+    Message.create!(chat: chat, user: sender, text: message, seen: other_user_connected?)
     payload = {
-      room_id: room_id, content: message, sender: sender, participants: convo.users
+      room_id: room_id, content: message, sender: sender, participants: chat.users
     }
     ActionCable.server.broadcast("ChatRoom-#{room_id}", payload)
   end
 
+  def unsubscribed
+    is_a = (current_user == chat.user_a)
+    is_a ? chat.update(user_a_connected: false) : chat.update(user_b_connected: false)
+  end
+
   private
 
   def get_chat(room_code)
     Chat.find(room_code)
   end
+
+  def other_user_connected?
+    current_user == chat.user_a && chat.user_b_connected ||
+      current_user == chat.user_a && chat.user_b_connected
```

Comment:
which is the difference between these two lines?
Maybe you wanted to do the same with the letters changed?

Reply:
Yes, should be the same but with the letters changed.